examples/dane/src/mastra/agents/index.ts

```diff
@@ -8,6 +8,55 @@ import { fsTool } from '../tools/fs.js';
 import { imageTool } from '../tools/image.js';
 import { readPDF } from '../tools/pdf.js';
 
+export const daneCommitMessage = new Agent({
+  name: 'DaneCommitMessage',
+  instructions: `
+    You are Dane, the ultimate GitHub operator.
+    You help engineers generate commit messages.
+
+    GENERATE MESSAGES ACCORDING TO THE GIT COMMIT MESSAGE CONVENTION: https://www.conventionalcommits.org/en/v1.0.0/
+
+    GENERATE A SCOPE FOR THE COMMIT MESSAGE IF NECESSARY, USING THE fsTool to UNDERSTAND THE FOLDER STRUCTURE.
+
+    THE SCOPE SHOULD BE THE PACKAGE NAME (llm, core, engine, etc), EXAMPLE NAME (e.g dane, crypto-chatbot, etc), THE INTEGRATION NAME (e.g github, etc).
+    IT CAN ALSO BE A LOGICAL MEMBER OF CORE (e.g memory, tools, workflows, etc).
+    BASICALLY FIGURE OUT THE BEST TOP LEVEL SEMANTIC MATCH FOR THE SCOPE.
+
+    DO NOT ATTEMPT TO USE GENERAL KNOWLEDGE! We are only as good as the tools we use.
```

Comment:
Is there some excess stuff in this prompt?

Reply:
and the tools?

Reply:
re: prompt
yeah, it's a bit much.. but it helped the agent be more discerning with figuring out the right scope for a staged change

re: tools
- provided the `fs` tool so the agent could read the directory structure.. to figure out scoping
- provided the `googleSearch` and `browser` tools so it could access the `conventional-commits` link i gave to it in the prompt

I'll refine it further